app/src/main/kotlin/com/wire/android/ui/home/conversations/selfdeletion/SelfDeletionMapper.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.android.ui.home.conversations.selfdeletion
+
+import com.wire.android.ui.home.messagecomposer.state.SelfDeletionDuration
+
+object SelfDeletionMapper {
+    fun Int?.toSelfDeletionDuration(): SelfDeletionDuration =
+        when (this) {
+            in 1..10 -> SelfDeletionDuration.TenSeconds
+            in 11..300 -> SelfDeletionDuration.FiveMinutes
+            in 301..3600 -> SelfDeletionDuration.OneHour
+            in 3601..86400 -> SelfDeletionDuration.OneDay
+            in 86401..604800 -> SelfDeletionDuration.OneWeek
+            in 604801..2592000 -> SelfDeletionDuration.FourWeeks
+            else -> SelfDeletionDuration.None
+        }
+
+    fun SelfDeletionDuration.toSeconds(): Int? =
+        when (this) {
+            SelfDeletionDuration.TenSeconds -> 10
+            SelfDeletionDuration.FiveMinutes -> 300
+            SelfDeletionDuration.OneHour -> 3600
+            SelfDeletionDuration.OneDay -> 86400
+            SelfDeletionDuration.OneWeek -> 604800
+            SelfDeletionDuration.FourWeeks -> 2592000
+            SelfDeletionDuration.None -> null
+        }
```

Comment:
@Suppress("MagicNumber")
enum class SelfDeletionDuration(val value: Duration?, val label: UIText) {
    None(null, UIText.StringResource(R.string.label_off)),
    TenSeconds(10.seconds, UIText.PluralResource(R.plurals.seconds_label, 10, 10)),
    FiveMinutes(5.minutes, UIText.PluralResource(R.plurals.minutes_label, 5, 5)),
    OneHour(1.hours, UIText.PluralResource(R.plurals.hours_label, 1, 1)),
    OneDay(1.days, UIText.PluralResource(R.plurals.days_label, 1, 1)),
    OneWeek(7.days, UIText.PluralResource(R.plurals.days_label, 7, 7)),
    FourWeeks(28.days, UIText.PluralResource(R.plurals.weeks_label, 4, 4))
}


No need to do this, we already can get info about the duration in more readable format using Duration

Reply:
True story bro 🎩 🍺